pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -0,0 +1,236 @@
+)
+
+var log = logf.Log.WithName("controller_nstemplatetier")
+
+const (
+	// maxPoolSize the maximum number of TemplateUpdateRequest resources that can exist at the same time
+	maxPoolSize = 5
+)
+
+// Add creates a new NSTemplateTier Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, config *configuration.Config) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileNSTemplateTier{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("nstemplatetier-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource NSTemplateTier
+	err = c.Watch(&source.Kind{Type: &toolchainv1alpha1.NSTemplateTier{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource TemplateUpdateRequest and requeue the owner NSTemplateTier
+	err = c.Watch(&source.Kind{Type: &toolchainv1alpha1.TemplateUpdateRequest{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &toolchainv1alpha1.NSTemplateTier{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileNSTemplateTier implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileNSTemplateTier{}
+
+// ReconcileNSTemplateTier reconciles a NSTemplateTier object
+type ReconcileNSTemplateTier struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client                 client.Client
+	scheme                 *runtime.Scheme
+	retrieveMemberClusters cluster.GetMemberClustersFunc
+}
+
+// Reconcile reads that state of the cluster for a NSTemplateTier object and makes changes based on the state read
+// and what is in the NSTemplateTier.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNSTemplateTier) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	logger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	logger.Info("Reconciling NSTemplateTier")
+
+	// Fetch the NSTemplateTier instance
+	instance := toolchainv1alpha1.NSTemplateTier{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, &instance)
+	if err != nil {
+		log.Error(err, "unable to get the current NSTemplateTier instance")
+		if errors.IsNotFound(err) {
```

Comment:
Move this check to line 96 (before logging the error).

Reply:
so what happens if there's an error which is not a `NotFound` one? Shouldn't we log the `err` as well?

Reply:
Yes. I would log all errors but NotFound.

Reply:
ok, thanks for the clarification

Reply:
done in 31aa93a